java/lib/src/main/java/com/svix/exceptions/ApiException.java

```diff
@@ -2,9 +2,44 @@
 
 import java.util.Map;
 import java.util.List;
+import com.google.gson.Gson;
+
+class APIError {
+    private String code;
+    private String detail;
+
+    public String getCode() {
+        return this.code;
+    }
+
+    public String getDetail() {
+        return this.detail;
+    }
+}
 
 public class ApiException extends com.svix.internal.ApiException {
+
+    private String message;
+
     public ApiException(final String message, final Throwable throwable, final int code, final Map<String, List<String>> responseHeaders, final String responseBody) {
         super(message, throwable, code, responseHeaders, responseBody);
+
+        if ((message == null || message.isEmpty()) && responseBody != null) {
+            try {
+                Gson gson = new Gson();
+                APIError error = gson.fromJson(responseBody, APIError.class);
+                this.message = error.getDetail();
```

Comment:
Maybe also show the code?

Reply:
Yeah we can do that, although at that point it might not make sense to parse the body at all and just print the whole thing as `message`?

`com.svix.exceptions.ApiException: Invalid token. (Code: "authentication_failed")`
vs
`com.svix.exceptions.ApiException: {"code":"authentication_failed","detail":"Invalid token."}`

What do you think?

Reply:
```
com.svix.exceptions.ApiException: authentication_failed: Invalid token.
```
How about?

Reply:
Actually, maybe your first suggestion or:
```
com.svix.exceptions.ApiException: Invalid token. (authentication_failed)
```
Or
```
com.svix.exceptions.ApiException: Invalid token. (code: authentication_failed)
```

Reply:
i like `com.svix.exceptions.ApiException: Invalid token. (authentication_failed)`